internal/cluster/transition.go

```diff
@@ -225,20 +236,30 @@ func getClusterMonitoringOperatorsStatus(cluster *common.Cluster) (bool, Monitor
 	}
 
 	for _, operator := range cluster.MonitoredOperators {
+		th.log.Debugf("cluster: %s, %s operator %s status is: %s ", cluster.ID.String(), operator.OperatorType, operator.Name, operator.Status)
 		operatorsStatuses[operator.OperatorType][operator.Status] = append(operatorsStatuses[operator.OperatorType][operator.Status], operator.Name)
 	}
+	th.log.Debugf("cluster: %s, progress: %+v ", cluster.ID.String(), cluster.Progress)
+	th.log.Debugf("cluster: %s, info: %+v ", cluster.ID.String(), cluster)
 
-	return haveBuiltinOperatorsComplete(operatorsStatuses[models.OperatorTypeBuiltin]) &&
-		haveOLMOperatorsComplete(operatorsStatuses[models.OperatorTypeOlm]), operatorsStatuses
+	return th.haveBuiltinOperatorsComplete(cluster.ID, operatorsStatuses[models.OperatorTypeBuiltin]) &&
+		th.haveOLMOperatorsComplete(cluster.ID, operatorsStatuses[models.OperatorTypeOlm]), operatorsStatuses
 }
 
-func haveBuiltinOperatorsComplete(operatorsNamesByStatus OperatorsNamesByStatus) bool {
-	// All the builtin operators are mandatory for a successfull installation
-	return len(operatorsNamesByStatus[models.OperatorStatusAvailable]) == countOperatorsInAllStatuses(operatorsNamesByStatus)
+func (th *transitionHandler) haveBuiltinOperatorsComplete(clusterID *strfmt.UUID, operatorsNamesByStatus OperatorsNamesByStatus) bool {
+	// All the builtin operators are mandatory for a successful installation
+	builtInOperatorsInAllStatusesCount := countOperatorsInAllStatuses(operatorsNamesByStatus)
+	th.log.Debugf("cluster: %s, OperatorStatusAvailable: %d, builtInOperatorsInAllStatusesCount is: %d",
+		clusterID.String(), len(operatorsNamesByStatus[models.OperatorStatusAvailable]), builtInOperatorsInAllStatusesCount)
+	return len(operatorsNamesByStatus[models.OperatorStatusAvailable]) == builtInOperatorsInAllStatusesCount
 }
 
-func haveOLMOperatorsComplete(operatorsNamesByStatus OperatorsNamesByStatus) bool {
+func (th *transitionHandler) haveOLMOperatorsComplete(clusterID *strfmt.UUID, operatorsNamesByStatus OperatorsNamesByStatus) bool {
 	// Need to wait for OLM operators to finish. Either available or failed. Failed operators would cause a degraded state.
+	OlmOperatorsInAllStatusesCount := countOperatorsInAllStatuses(operatorsNamesByStatus)
+	th.log.Debugf("cluster: %s, OperatorStatusAvailable + OperatorStatusFailed: %d, builtInOperatorsInAllStatusesCount is: %d",
+		clusterID.String(), len(operatorsNamesByStatus[models.OperatorStatusAvailable])+len(operatorsNamesByStatus[models.OperatorStatusFailed]),
+		OlmOperatorsInAllStatusesCount)
 	return len(operatorsNamesByStatus[models.OperatorStatusAvailable])+len(operatorsNamesByStatus[models.OperatorStatusFailed]) ==
```

Comment:
The corresponding logic was changed (not counting failed for OLM operators), but did not change here.
The reason is that it is a logic duplication.  We should have it in single place.

Reply:
This area could use a refactor (which, as we discussed verbally - should not take place in this PR).
The change of behavior here is intentional (commit-msg). We want the cluster to be marked as "installed" even if it is degraded with a case of a failed OLM operator. We do not want to increment progress value during that time to avoid confusing our users while it retries. See more details in the commit-msg.